src/main/java/de/tum/in/www1/artemis/web/rest/tutorialgroups/TutorialGroupSessionResource.java

```diff
@@ -152,6 +154,36 @@ public ResponseEntity<TutorialGroupSession> update(@PathVariable Long courseId,
         return ResponseEntity.ok(TutorialGroupSession.preventCircularJsonConversion(result));
     }
 
+    /**
+     * PUT /courses/:courseId/tutorial-groups/:tutorialGroupId/sessions/:sessionId/attendance-count : Updates the attendance count of a tutorial group session
+     *
+     * @param courseId                               the id of the course to which the tutorial group belongs to
+     * @param tutorialGroupId                        the id of the tutorial group to which the session belongs to
+     * @param sessionId                              the id of the session to update
+     * @param tutorialGroupSessionAttendanceCountDTO DTO containing the updated attendance count
+     * @return the ResponseEntity with status 200 (OK) and with body the updated tutorial group session
+     */
+    @PutMapping("/courses/{courseId}/tutorial-groups/{tutorialGroupId}/sessions/{sessionId}/attendance-count")
+    @PreAuthorize("hasRole('TA')")
+    @FeatureToggle(Feature.TutorialGroups)
+    public ResponseEntity<TutorialGroupSession> updateAttendanceCount(@PathVariable Long courseId, @PathVariable Long tutorialGroupId, @PathVariable Long sessionId,
+            @RequestBody @Valid TutorialGroupSessionAttendanceCountDTO tutorialGroupSessionAttendanceCountDTO) {
+        log.debug("REST request to update attendance count of session: {} of tutorial group: {} of course {} to {}", sessionId, tutorialGroupId, courseId,
+                tutorialGroupSessionAttendanceCountDTO.attendanceCount());
+        var sessionToUpdate = this.tutorialGroupSessionRepository.findByIdElseThrow(sessionId);
+        checkEntityIdMatchesPathIds(sessionToUpdate, Optional.of(courseId), Optional.of(tutorialGroupId), Optional.of(sessionId));
+        tutorialGroupService.isAllowedToModifySessionsOfTutorialGroup(sessionToUpdate.getTutorialGroup(), null);
+        sessionToUpdate.setAttendanceCount(tutorialGroupSessionAttendanceCountDTO.attendanceCount());
+        var result = tutorialGroupSessionRepository.save(sessionToUpdate);
+        return ResponseEntity.ok(TutorialGroupSession.preventCircularJsonConversion(result));
+    }
+
+    /**
+     * DTO for updating the attendance count of a tutorial group session
+     */
+    public record TutorialGroupSessionAttendanceCountDTO(@Min(0) @Max(999) Integer attendanceCount) {
```

Comment:
I'd change this to somewhere around 3000, in case anyone wants to use this for a lecture with many students

Reply:
And why do we need a DTO for only one value, can't we send the value directly?

Reply:
I agree, just send the value directly. You could move the min/max annotation to the domain class.

Reply:
Done in 1ff34058ad75b40096aef59dd661a304421d18b1